bundles/org.openhab.binding.withings/src/main/java/org/openhab/binding/withings/internal/handler/WithingsBridgeHandler.java

```diff
@@ -0,0 +1,239 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Sven Strohschein - Initial contribution
+ */
+@NonNullByDefault
+public class WithingsBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(WithingsBridgeHandler.class);
+
+    private final HttpClient httpClient;
+    private final AccessTokenInitializableService accessTokenService;
+    private final List<WithingsThingHandler> thingHandlers;
+    public WithingsBridgeConfiguration configuration = new WithingsBridgeConfiguration();
+    private @Nullable ScheduledFuture<?> tokenRefreshJob;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private Optional<WithingsDataModel> currentModel;
+
+    public WithingsBridgeHandler(Bridge bridge, AccessTokenInitializableService accessTokenService,
+            HttpClient httpClient) {
+        super(bridge);
+        this.accessTokenService = accessTokenService;
+        this.httpClient = httpClient;
+        this.thingHandlers = new ArrayList<>();
+        this.currentModel = Optional.empty();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Withings API bridge handler.");
+
+        configuration = getConfigAs(WithingsBridgeConfiguration.class);
+
+        scheduleTokenInitAndRefresh();
+    }
+
+    public void registerThingHandler(WithingsThingHandler thingHandler) {
+        thingHandlers.add(thingHandler);
+
+        notifyThingHandler(thingHandler);
+    }
+
+    private void connectionSucceed() {
+        updateData();
+        scheduleRefreshJob();
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private void scheduleTokenInitAndRefresh() {
+        tokenRefreshJob = scheduler.scheduleWithFixedDelay(() -> {
+            logger.debug("Initializing API Connection and scheduling token refresh every {}s",
+                    WithingsBindingConstants.TOKEN_REFRESH_SECONDS);
+            try {
+                accessTokenService.init(thing.getUID().getAsString(), configuration);
+
+                boolean isAuthCodeRedeemed = processAuthCodeIfRequired();
+
+                if (!isAuthCodeRedeemed) {
+                    Optional<String> refreshToken = accessTokenService.getRefreshToken();
+                    refreshToken.ifPresent(this::refreshAccessToken);
+                }
+
+                if (accessTokenService.getAccessToken().isPresent()) {
+                    connectionSucceed();
+                } else {
+                    logger.warn("There is no valid access token! Please configure a new auth-code.");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "Please configure a new auth-code");
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Same for below.

Reply:
Hm and when someone like me disabled all event logs because the amount (of some bindings) is just too high? ;-)

Reply:
You can set the log level Java package-based. Also you can set the log level of Item state changes and Thing state changes independently. Also it seems this error is persistent, so the user will see it in the UI.

Reply:
Are you sure? I think the problem is that events are logged with the logger "log4j2.logger.smarthomeItemStateEvent" independent from the binding which causes the event / item state change. Therefore it can only get switched on or off by changing the logging level of this single logger. The logging level of package-based logger of the binding has no effect for events.log. Something similar is described here: https://github.com/openhab/openhab-core/issues/1025

Reply:
That's correct, you can't set the log level of ThingStatusInfoEvent per binding. But, again, you can set `ThingStatusInfoEvent` independently from `ItemStateEvent`, regarding your argument with the thermostat updates.